src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -762,8 +764,14 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
             String ucCredentialsId = uc.getCredentialsId();
             if (ucCredentialsId != null) {
                 String url = getParameterString(uc.getUrl(), environment);
-                List<StandardUsernameCredentials> urlCredentials = CredentialsProvider.lookupCredentials(StandardUsernameCredentials.class, project,
-                                        ACL.SYSTEM, URIRequirementBuilder.fromUri(url).build());
+                List<StandardUsernameCredentials> urlCredentials = CredentialsProvider.lookupCredentials(
+                        StandardUsernameCredentials.class,
+                        project,
+                        project instanceof Queue.Task
+                                ? Tasks.getDefaultAuthenticationOf((Queue.Task)project)
+                                : ACL.SYSTEM,
+                        URIRequirementBuilder.fromUri(url).build()
+                );
                 CredentialsMatcher ucMatcher = CredentialsMatchers.withId(ucCredentialsId);
                 CredentialsMatcher idMatcher = CredentialsMatchers.allOf(ucMatcher, GitClient.CREDENTIALS_MATCHER);
```

Comment:
While I'm at it, @MarkEWaite we should probably look to move to using AuthenticationTokens (see https://github.com/jenkinsci/credentials-plugin/blob/master/docs/consumer.adoc#retrieve-the-secret-from-a-specific-credentials-instance) rather than using a hard-coded collection... but that should be in a separate JIRA issue and with a bump to 3.7.x

Reply:
That would be great with me.

Switching to authentication tokens would be a great user-valuable reason to increment the version number to 4.0.

I've been preparing for the 4.0 release that will switch to Java 8.  The git client plugin switch has been in progress on my java-8 branch for a while.  The upgrade to JGit 4.9 broke some tests, so I had to "fall back" to JGit 4.8 until I can investigate  the test failures from JGit 4.9.  The git plugin transition to java-8 only began on a private branch a week or two ago.  I hope to complete them both before the end of calendar 2017.  It's more than time to switch to Java 8...